cmd/heartbeat/heartbeat.go

```diff
@@ -76,6 +78,16 @@ func SendHeartbeats(v *viper.Viper, queueFilepath string) error {
 	setLogFields(params)
 	log.Debugf("params: %s", params)
 
+	if RateLimited(params.Offline) {
+		var rateLimitedErr error
+		if rateLimitedErr = offlinecmd.SaveHeartbeats(v, nil, queueFilepath); rateLimitedErr == nil {
+			return nil
+		}
```

Comment:
```suggestion
		if err := offlinecmd.SaveHeartbeats(v, nil, queueFilepath); err == nil {
			return nil
		}
```

Reply:
We need the `err` in the line below for logging, but that `err` is actually from:
https://github.com/wakatime/wakatime-cli/pull/1071/files#diff-a42970af11c59a3cbee6b9abd656868ab6508da9003c66d042bcba1f0514811aR72

Because of scoping, so it's not actually the `err` we're looking for.

Reply:
My suggestion makes `err` scoped to `if` statement. @alanhamlett

Reply:
Yea but we need that err outside the if statement 2 lines down.

Reply:
I'll modify the code to be more clear.